iset.mm

```diff
@@ -64121,6 +64129,112 @@ all reals (greatest real number) for which all positive reals are greater.
   $}
 
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Omniscient sets
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c Omni $.  $( Class of all omniscient sets. $)
+
+  $( Extend class definition to include the class of omniscient sets. $)
+  comni $a class Omni $.
+
+  ${
+    $d f x y $.
+    $( An omniscient set is one where we can decide whether a predicate (here
+       represented by a function ` f ` ) holds (is equal to ` 1o ` ) for all
+       elements or fails to hold (is equal to ` (/) ` ) for some element.
+       Definition 3.1 of [Pierik], p. 14.
+
+       In particular, ` _om e. Omni ` is known as the Lesser Principle of
+       Omniscience (LPO).  (Contributed by Jim Kingdon, 28-Jun-2022.) $)
+    df-omni $a |- Omni = { y | A. f ( f : y --> 2o
+        -> ( E. x e. y ( f ` x ) = (/) \/ A. x e. y ( f ` x ) = 1o ) ) } $.
+  $}
+
+  ${
+    $d A f x y $.
+    $( Expansion of ~ df-omni in terms of classes.  (Contributed by Jim
+       Kingdon, 28-Jun-2022.) $)
+    dfomnicl $p |- ( A e. V -> ( A e. Omni <-> A. f ( f : A --> 2o
```

Comment:
Isn't our convention to use "isomni", with comment "The predicate of being omniscient." or something like that ?

Side remark: the statement you wrote is the most useful statement, but it is a characterization for sets, not for classes. The characterization for classes is
> |- ( A e. Omni <-> ( A e. _V /\ A. f ... ) )

Sometimes, both statements are in set.mm, but generally the one you wrote is more useful.

Reply:
I agree with @benjub that the name of this theorem should be changed.

Reply:
I was hunting around for existing conventions (because I had forgot about the `isfoo` thing) and didn't find it. So thanks for this.

As for class versus set, I suppose that comment could have been worded as "class variable" rather than "class" but I'll just go with "The predicate of being omniscient." which avoids the issue.